src/org/traccar/notification/EventForwarder.java

```diff
@@ -15,54 +15,75 @@
  */
 package org.traccar.notification;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.ning.http.client.AsyncHttpClient.BoundRequestBuilder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 import org.traccar.Context;
 import org.traccar.helper.Log;
 import org.traccar.model.Device;
 import org.traccar.model.Event;
 import org.traccar.model.Geofence;
 import org.traccar.model.Position;
 
-import java.nio.charset.StandardCharsets;
-import java.util.HashMap;
-import java.util.Map;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.ning.http.client.AsyncHttpClient.BoundRequestBuilder;
+import com.ning.http.client.FluentCaseInsensitiveStringsMap;
 
-public final class EventForwarder {
+public abstract class EventForwarder {
 
     private String url;
     private String header;
+    private final String payloadParamName;
+    private final String additionalParams;
 
     public EventForwarder() {
         url = Context.getConfig().getString("event.forward.url", "http://localhost/");
         header = Context.getConfig().getString("event.forward.header", "");
+        payloadParamName = Context.getConfig().getString("event.forward.paramMode.payloadParamName", "payload");
```

Comment:
Please shorten these two config keys.

Reply:
i thought that could be a good idea to use that name because they indicate that are exlusive params for the 'paramMode' and they will have no effect using the existing 'json' mode (or another one) of 'event.forward', so leaving something like 'event.forward.payloadParamName' could lead to confusion imho.

Reply:
OK, if it's only for paramMode what is it doing in the base class then?

Reply:
yes you are right, this no longer makes sense to be on the base class, fixed.